src/registrar/admin.py

```diff
@@ -328,29 +384,57 @@ def save_model(self, request, obj, form, change):
                 # Get the original application from the database
                 original_obj = models.DomainApplication.objects.get(pk=obj.pk)
 
-                if obj.status != original_obj.status:
-                    status_method_mapping = {
-                        models.DomainApplication.STARTED: None,
-                        models.DomainApplication.SUBMITTED: obj.submit,
-                        models.DomainApplication.IN_REVIEW: obj.in_review,
-                        models.DomainApplication.ACTION_NEEDED: obj.action_needed,
-                        models.DomainApplication.APPROVED: obj.approve,
-                        models.DomainApplication.WITHDRAWN: obj.withdraw,
-                        models.DomainApplication.REJECTED: obj.reject,
-                        models.DomainApplication.INELIGIBLE: obj.reject_with_prejudice,
-                    }
-                    selected_method = status_method_mapping.get(obj.status)
-                    if selected_method is None:
-                        logger.warning("Unknown status selected in django admin")
-                    else:
-                        # This is an fsm in model which will throw an error if the
-                        # transition condition is violated, so we roll back the
-                        # status to what it was before the admin user changed it and
-                        # let the fsm method set it.
-                        obj.status = original_obj.status
-                        selected_method()
-
-            super().save_model(request, obj, form, change)
+                if (
+                    obj
+                    and original_obj.status == models.DomainApplication.APPROVED
+                    and (
+                        obj.status == models.DomainApplication.REJECTED
+                        or obj.status == models.DomainApplication.INELIGIBLE
+                    )
+                    and not obj.domain_is_not_active()
```

Comment:
Apologies if I'm missing context (and this may be a NIT), but it may be preferred to have a `domain_is_active` and then do a `not` against that instead of currently having `not` against another 'not active` which can be confusing.

However I do see below that you're referencing `domain_is_not_active` a lot in `src/registrar/models/domain_application.py` ...just some food for thought 😅

(Let me know if this doesn't make sense, I can reword or hop on call to explain!)

Reply:
Agreed I am curious why is active was changed here. It is also a requirement for ticket 848 so changing it here changes that ticket.

Reply:
Yep, conditions=[not domain_is_active] returns this error.

![Screen Shot 2023-09-11 at 2 29 21 PM](https://github.com/cisagov/getgov/assets/107004823/9c27dceb-cf6b-4c16-92b8-24face3b23a0)